Add warning to payment method edit page

luisramos0 opened the pull request:
#### What? Why?

Following #5424 (shipping methods), in this PR we add a warning message to payment methods edit page as well.

![image](https://user-images.githubusercontent.com/1640378/81833128-3c8f8500-9537-11ea-90f2-cb4989ce289f.png)

As part of this PR I tested that the display field is working for both "Checkout only" and "Backoffice only". This PR makes these option values translatable!

Also, we bring quite a few translations that were only in spree in the payments list, new and edit page. Looking at FR production I can see the form labels Tags and Active in the edit page will now be translatable.

#### What should we test?
Verify the new warning and check translations are not broken.
Verify the "display" field appears for both managers and super admins.

#### Release notes
Changelog Category: Added
Add warning message to payment methods edit page so that user is aware of the problems of de-activating a payment method and knows what is the alternative.

sauloperez commented:
@luisramos0 you'll need to rebase to get the fix on the job logger spec.

luisramos0 commented:
:+1: rebased

filipefurtad0 commented:
Hey @luisramos0,
The branch has conflicts which need attention I think.
Thank you.

luisramos0 commented:
yes, thanks Filipe.
rebased to resolve conflicts.
The conflicts are related to making the display field also available for managers (something done in a recent PR), I added a note to "what to test?" above related to this :+1:

filipefurtad0 commented:
Hi again @luisramos0 ,

I can't seem to observe the warning in the payment method edit page. Please see below the Shipping Methods and the Payment Methods page, after staging. In this regard, an error was during/after staging, which may relate to the fact that the PR is not doing what it is expected:

https://openfoodnetwork.slack.com/archives/CEF14NU3V/p1591109733301800 

![image](https://user-images.githubusercontent.com/49817236/83538738-1d5c9580-a4ee-11ea-83d1-755d55edb907.png)

luisramos0 commented:
Something wrong with the staging server. This is what I was seeing:
![image](https://user-images.githubusercontent.com/1640378/83549746-b6df7380-a4fd-11ea-8d88-e6d94f22cb96.png)

ping @Matt-Yorkley fyi
I provisioned the server and re-deployed the PR, it is still broken.
I raised an issue for it: https://github.com/openfoodfoundation/ofn-install/issues/633
I cant find the solution for the issue and I dont want to fix it manually in staging ES so other devs will see it broken and investigate. So, for now, let's freeze staging ES for ofn-install#633

filipefurtad0 commented:
Thank you for the update @luisramos0, and for raising the issue.

filipefurtad0 commented:
Hi @luisramos0 ,

Staging worked now, and the changes this PR introduces are observable:

- Manager
![image](https://user-images.githubusercontent.com/49817236/83681723-b23cbd00-a5da-11ea-9479-6ae300b19533.png)

- Superadmin
![image](https://user-images.githubusercontent.com/49817236/83682623-157b1f00-a5dc-11ea-961e-1e3bcb827021.png)

Translations seem good too.
Ready to go!

luisramos0 commented:
:ok_hand:
